bindings/node.js/kzg.cxx

```diff
@@ -152,190 +152,215 @@ Napi::Value BlobToKzgCommitment(const Napi::CallbackInfo& info) {
   return napi_typed_array_from_bytes((uint8_t *)(&commitment), BYTES_PER_COMMITMENT, env);
 }
 
-// computeAggregateKzgProof: (blobs: Blob[], setupHandle: SetupHandle) => KZGProof;
-Napi::Value ComputeAggregateKzgProof(const Napi::CallbackInfo& info) {
+// computeKzgProof: (blob: Blob, zBytes: Bytes32, setupHandle: SetupHandle) => KZGProof;
+Napi::Value ComputeKzgProof(const Napi::CallbackInfo& info) {
   auto env = info.Env();
 
   size_t argument_count = info.Length();
-  size_t expected_argument_count = 2;
+  size_t expected_argument_count = 3;
   if (argument_count != expected_argument_count) {
     return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
   }
 
-  auto blobs_param = info[0].As<Napi::Array>();
-  auto kzg_settings = info[1].As<Napi::External<KZGSettings>>().Data();
-
-  auto blobs_count = blobs_param.Length();
-
-  auto blobs = (Blob*)calloc(blobs_count, sizeof(Blob));
-  if (blobs == NULL) {
-    Napi::Error::New(env, "Error while allocating memory for blobs").ThrowAsJavaScriptException();
-    return env.Null();
-  };
-
-  for (uint32_t blob_index = 0; blob_index < blobs_count; blob_index++) {
-    Napi::Value blob = blobs_param[blob_index];
-    auto blob_bytes = blob.As<Napi::Uint8Array>().Data();
-    memcpy(blobs[blob_index].bytes, blob_bytes, BYTES_PER_BLOB);
-  }
+  auto blob = extract_byte_array_from_param(info, 0, "blob");
+  auto z_bytes = extract_byte_array_from_param(info, 1, "zBytes");
+  auto kzg_settings = info[2].As<Napi::External<KZGSettings>>().Data();
 
   KZGProof proof;
-  C_KZG_RET ret = compute_aggregate_kzg_proof(
+  C_KZG_RET ret = compute_kzg_proof(
     &proof,
-    blobs,
-    blobs_count,
+    (Blob *)blob,
+    (Bytes32 *)z_bytes,
     kzg_settings
   );
-  free(blobs);
 
   if (ret != C_KZG_OK) {
-     Napi::Error::New(env, "Failed to compute aggregated proof")
+     Napi::Error::New(env, "Failed to compute proof")
       .ThrowAsJavaScriptException();
     return env.Undefined();
   };
 
   return napi_typed_array_from_bytes((uint8_t *)(&proof), BYTES_PER_PROOF, env);
 }
 
-// computeKzgProof: (blob: Blob, zBytes: Bytes32, setupHandle: SetupHandle) => KZGProof;
-Napi::Value ComputeKzgProof(const Napi::CallbackInfo& info) {
+// computeBlobKzgProof: (blob: Blob, setupHandle: SetupHandle) => KZGProof;
+Napi::Value ComputeBlobKzgProof(const Napi::CallbackInfo& info) {
   auto env = info.Env();
 
   size_t argument_count = info.Length();
-  size_t expected_argument_count = 3;
+  size_t expected_argument_count = 2;
   if (argument_count != expected_argument_count) {
     return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
   }
 
   auto blob = extract_byte_array_from_param(info, 0, "blob");
-  auto z_bytes = extract_byte_array_from_param(info, 1, "zBytes");
-  auto kzg_settings = info[2].As<Napi::External<KZGSettings>>().Data();
+  auto kzg_settings = info[1].As<Napi::External<KZGSettings>>().Data();
 
```

Comment:
Same as in https://github.com/ethereum/c-kzg-4844/pull/147/files#r1114171554

Reply:
Fixed.